File: Assets/MoveCube.py
class MoveCube(Actor.Actor):
    def __init__(self):
        self.CubeContainer = Container(0)  #Container
        self.test1 = Container          # type
        self.intData = int                # type
        self.intData2 = int(0)            # int
        return
    def OnCreate(self, uid):
        self.CubeComponent = self.CubeContainer.FindComponentByType("Cube")
        self.CubeTransComponent = self.CubeContainer.FindComponentByType("TransformGroup")
        self.CubeComponent.PropMaterial.SetTextureDiffuse("$project/Assets/1vf.png")
        self.CubeTransComponent.SetPosition(Math3d.Vector3(0,1,0))
        
        return 0 
    def OnDestroy(self):
        return 0 
    def OnEnable(self):
        return 0 
    def OnDisable(self):
        return 0 
    def Update(self):
        return

    def OnMessage(self, msg, number, Vector4_lparm, Vector4_wparam):
        return;


   

File: Assets/openex/MoveCube.cs
public class MoveCube : EActor
{
    public Container CubeContainer;

    private Cube cubeComponent;

    public int OnCreate()
    {
        //cubeComponent = (Cube)CubeContainer.FindComponentByType("Cube");
        //if (cubeComponent == null)
        //{
        //    Log("cubeComponent is null", 0, 0.0);
        //    return 0;
        //}

    }

    //public void ChangeCubeTexture()
    //{
    //    if(cubeComponent != null)
    //    {
    //        cubeComponent.PropMaterial[0].SetTextureDiffuse(GetProjectPath() + "/Assets/Resource/example/1vf.png");
    //    }
        
    //}


    public override int OnMessage(string msg, int arg0, Vector4 arg1, Vector4 arg2)
    {

        
        return 0;
    }

    public override int Update()
    {


        return 0;
    }
}